Microsoft.Toolkit.Uwp/Helpers/ObjectStorage/ApplicationDataStorageHelper.cs

```diff
@@ -0,0 +1,313 @@
+        public void Delete(string compositeKey, string key)
+        {
+            if (!KeyExists(compositeKey))
+            {
+                throw new KeyNotFoundException($"Composite key not found: {compositeKey}");
+            }
+
+            ApplicationDataCompositeValue composite = (ApplicationDataCompositeValue)Settings.Values[compositeKey];
+
+            if (!composite.Remove(key))
+            {
+                throw new KeyNotFoundException($"Settings key not found: {key}");
+            }
+        }
+
+        /// <inheritdoc />
+        public Task<T> ReadFileAsync<T>(string filePath, T @default = default)
+        {
+            return ReadFileAsync<T>(Folder, filePath, @default);
+        }
+
+        /// <inheritdoc />
+        public Task<IEnumerable<(DirectoryItemType, string)>> ReadFolderAsync(string folderPath)
+        {
+            return ReadFolderAsync(Folder, folderPath);
+        }
+
+        /// <inheritdoc />
+        public Task CreateFileAsync<T>(string filePath, T value)
+        {
+            return SaveFileAsync<T>(Folder, filePath, value);
+        }
+
+        /// <inheritdoc />
+        public Task CreateFolderAsync(string folderPath)
+        {
+            return CreateFolderAsync(Folder, folderPath);
+        }
+
+        /// <inheritdoc />
+        public Task DeleteItemAsync(string itemPath)
+        {
+            return DeleteItemAsync(Folder, itemPath);
+        }
+
+        /// <summary>
+        /// Saves an object inside a file.
+        /// </summary>
+        /// <typeparam name="T">Type of object saved.</typeparam>
+        /// <param name="filePath">Path to the file that will contain the object.</param>
+        /// <param name="value">Object to save.</param>
+        /// <returns>Waiting task until completion.</returns>
+        public Task<StorageFile> SaveFileAsync<T>(string filePath, T value)
+        {
+            return SaveFileAsync<T>(Folder, filePath, value);
+        }
+
+        private async Task<T> ReadFileAsync<T>(StorageFolder folder, string filePath, T @default = default)
+        {
+            string value = await StorageFileHelper.ReadTextFromFileAsync(folder, filePath);
+            return (value != null) ? Serializer.Deserialize<T>(value) : @default;
+        }
+
+        private async Task<IEnumerable<(DirectoryItemType, string)>> ReadFolderAsync(StorageFolder folder, string folderPath)
+        {
+            var targetFolder = await folder.GetFolderAsync(folderPath);
+            var items = await targetFolder.GetItemsAsync();
+
+            return items.Select((item) =>
+            {
+                var itemType = item.IsOfType(StorageItemTypes.File) ? DirectoryItemType.File
+                    : item.IsOfType(StorageItemTypes.Folder) ? DirectoryItemType.Folder
+                    : DirectoryItemType.None;
+
+                return new ValueTuple<DirectoryItemType, string>(itemType, item.Name);
+            }).ToList();
```

Comment:
We don't need to eagerly populate the collection here as we're returning `IEnumerable<T>`.
Also even if we wanted to return a fixed-size collection, `ToList()` wouldn't be the best choice here anyway and we should've used `ToArray()` instead. In this case I'd suggest to just remove that `ToList()` entirely 👍